### PR TITLE
WIP: link the Strategy and Order managers

### PR DESCRIPTION
**Pull request by DCPRevere**

### CODE REVIEW

File: src/price/service.price.platform.hpp [hunk 1]
```diff
@@ -31,16 +31,28 @@ namespace antara::mmbot
 {
     using registry_price_result = std::unordered_map<antara::pair, st_price>;
 
-    class price_service_platform
+    class abstract_price_service_platform
```
Comment (Milerius) [inline src/price/service.price.platform.hpp]: I really disagree as using abstract class for price_service_platform (there is only one service)

Use template class instead

File: src/strategy_manager/strategy.manager.tests.cpp [hunk 2]
```diff
@@ -32,9 +35,29 @@ namespace antara::mmbot::tests
...
+    }
+
     TEST_CASE ("strats can be added and retreived")
     {
-        auto sm = strategy_manager();
+        dex dex;
+        cex cex;
+        order_manager_mock om(dex, cex);
+        auto ps = price_service_platform_mock();
+        auto sm = strategy_manager(ps, om);
```
Comment (Milerius) [inline src/strategy_manager/strategy.manager.tests.cpp]: strategy_manager can be templated on the implementation class such as:

`strategy_manager<price_service_platform_fake>(ps, om);`

Comment (Milerius) [reply src/strategy_manager/strategy.manager.tests.cpp]: Abstract classes and interfaces are useful when you have to manipulate several, not when you have to mock a class. The templates will do the trick here

**Review by Milerius — CHANGES_REQUESTED** [review]

**Comment by DCPRevere** [comment]
@Milerius I've made the change to use a template instead of the abstract class.

**Pull request merged**